src/Compilers/CSharp/Portable/Binder/Binder_Conversions.cs

```diff
@@ -1285,6 +1285,85 @@ internal bool HasCollectionExpressionApplicableAddMethod(SyntaxNode syntax, Type
             }
         }
 
+        internal bool HasCollectionExpressionAddMethod(SyntaxNode syntax, TypeSymbol targetType)
+        {
+            const string methodName = "Add";
+            var useSiteInfo = CompoundUseSiteInfo<AssemblySymbol>.Discarded;
+
+            var lookupResult = LookupResult.GetInstance();
+            LookupInstanceMember(lookupResult, targetType, leftIsBaseReference: false, rightName: methodName, rightArity: 0, invoked: true, ref useSiteInfo);
+            bool anyApplicable = lookupResult.IsMultiViable &&
+                lookupResult.Symbols.Any(s => s is MethodSymbol m && isApplicableAddMethod(m, expectingExtensionMethod: false));
+            lookupResult.Free();
+            if (anyApplicable)
+            {
+                return true;
+            }
+
+            var implicitReceiver = new BoundObjectOrCollectionValuePlaceholder(syntax, isNewInstance: true, targetType) { WasCompilerGenerated = true };
+            foreach (var scope in new ExtensionMethodScopes(this))
+            {
+                var methodGroup = MethodGroup.GetInstance();
+                PopulateExtensionMethodsFromSingleBinder(scope, methodGroup, syntax, implicitReceiver, rightName: methodName, typeArgumentsWithAnnotations: default, BindingDiagnosticBag.Discarded);
+                anyApplicable = methodGroup.Methods.Any(m => isApplicableAddMethod(m, expectingExtensionMethod: true));
+                methodGroup.Free();
+                if (anyApplicable)
+                {
+                    return true;
+                }
+            }
+
+            return false;
+
+            static bool isApplicableAddMethod(MethodSymbol method, bool expectingExtensionMethod)
+            {
+                if (method.IsStatic != expectingExtensionMethod)
+                {
+                    return false;
+                }
+
+                var parameters = method.Parameters;
+                int valueIndex = expectingExtensionMethod ? 1 : 0;
+                int requiredLength = valueIndex + 1;
+                if (parameters.Length < requiredLength)
+                {
+                    return false;
+                }
+
+                // Any trailing parameters must be optional or params.
+                for (int i = requiredLength; i < parameters.Length; i++)
+                {
+                    if (parameters[i] is { IsOptional: false, IsParams: false })
```

Comment:
>IsParams [](http://example.com/codeflow?start=62&length=8)

The ```IsParams``` check is inconsistent with how overload resolution does that.
- Only value on corresponding least overridden method matters
- The definition type of the least overridden method parameter matters, see ```OverloadResolution.IsValidParams```. In fact I suggest to use that helper here